controller/api/public/stat_summary.go

```diff
@@ -128,37 +131,50 @@ func statSummaryError(req *pb.StatSummaryRequest, message string) *pb.StatSummar
 	}
 }
 
-func (s *grpcServer) resourceQuery(ctx context.Context, req *pb.StatSummaryRequest) resourceResult {
+func (s *grpcServer) getKubernetesObjectStats(req *pb.StatSummaryRequest) (map[pb.Resource]k8sStat, error) {
```

Comment:
This is a valiant effort to reuse all of the existing k8s resource code, but I think that authorities are different enough from k8s resources that we'd be better off just forking to a different method entirely in the case of authorities. I recommend leaving the `resourceQuery` method mostly in tact, but renaming it to `k8sResourceQuery` or something similar. And then we could add a `nonK8sResourceQuery` method that looks like this:

```
func (s *grpcServer) nonK8sResourceQuery(ctx context.Context, req *pb.StatSummaryRequest) resourceResult {
	rows := make([]*pb.StatTable_PodGroup_Row, 0)

	requestMetrics, err := s.getPrometheusMetrics(ctx, req, req.TimeWindow)
	if err != nil {
		return resourceResult{err: err}
	}

	for key, stats := range requestMetrics {
		resource := key
		row := pb.StatTable_PodGroup_Row{
			Resource:   &resource,
			TimeWindow: req.TimeWindow,
			Stats:      stats,
		}

		rows = append(rows, &row)
	}

	rsp := pb.StatTable{
		Table: &pb.StatTable_PodGroup_{
			PodGroup: &pb.StatTable_PodGroup{
				Rows: rows,
			},
		},
	}

	return resourceResult{res: &rsp}
}
```

And then above, on line 96 where we fire off the queries in separate goroutines, we can do something like:

```
		go func() {
			if resource == k8s.Authority {
				resultChan <- s.nonK8sResourceQuery(ctx, statReq)
			} else {
				resultChan <- s.k8sResourceQuery(ctx, statReq)
			}
		}()
```

Reply:
Yeah! You're right, just splitting these out into separate cases makes them easier to follow, and we don't have to keep doing the k8s/nonk8s checks everywhere